crates/proof-sdk/proof/src/altda/altda_provider.rs

```diff
@@ -0,0 +1,58 @@
+use alloc::boxed::Box;
+use alloc::sync::Arc;
+use alloy_primitives::Bytes;
+use async_trait::async_trait;
+use kona_derive::traits::AltDAProvider;
+use kona_preimage::CommsClient;
+
+use crate::errors::OracleProviderError;
+
+use super::OracleEigenDAProvider;
+
+#[derive(Debug, Clone)]
+pub struct OracleAltDAProvider<T: CommsClient> {
+    /// The oracle eigenda provider.
+    eigenda_provider: OracleEigenDAProvider<T>,
+}
+
+impl<T: CommsClient> OracleAltDAProvider<T> {
+    /// Constructs a new oracle-backed AltDA provider.
+    pub fn new(eigenda_provider: OracleEigenDAProvider<T>) -> Self {
+        Self { eigenda_provider }
+    }
+
+    /// Constructs a new oracle-backed AltDA provider by constructing
+    /// the respective altda providers using the oracle.
+    pub fn new_from_oracle(oracle: Arc<T>) -> Self {
+        Self { eigenda_provider: OracleEigenDAProvider::new(oracle) }
+    }
+}
+
+#[async_trait]
+impl<T: CommsClient + Send + Sync> AltDAProvider for OracleAltDAProvider<T> {
+    type Error = OracleProviderError;
+    /// Retrieves a blob from the oracle.
+    ///
+    /// ## Takes
+    /// - `block_ref`: The block reference.
```

Comment:
what is block_ref here?

Reply:
Outdated! [f263bf6](https://github.com/Layr-Labs/kona/pull/1/commits/f263bf6003b968d69e9476ca761646b606419999)